apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/index.tsx

```diff
@@ -14,7 +19,40 @@ export function TradeWidget(props: TradeWidgetProps) {
     tradeQuoteStateOverride,
     enableSmartSlippage,
   } = params
-  const modals = TradeWidgetModals({confirmModal, genericModal, selectTokenWidget: slots.selectTokenWidget})
+  const modals = TradeWidgetModals({ confirmModal, genericModal, selectTokenWidget: slots.selectTokenWidget })
+
+  const { isVisible, onDismiss, isLimitOrdersTab } = useLimitOrdersPromoBanner()
+  const navigate = useNavigate()
+
+  const handleCtaClick = () => {
```

Comment:
I think it makes sense to add a smart component as a wrapper around `LimitOrdersPromoBanner` which will define `handleCtaClick` inside

Reply:
I see that you need to decide if the banner should be shown and display another content if not.
You can pass the fallback content as:
```
<LimitOrdersPromoBanner>
   {slots.topContent}
</LimitOrdersPromoBanner>
```

And inside of the compontent take a decision based on `isVisible` value

Reply:
Thanks for checking this. Refactored this now.